src/data-planes/views/DataPlaneListView.vue

```diff
@@ -0,0 +1,654 @@
+      return columnsDropdownItems
+        .filter((item) => this.multicluster ? true : item.tableHeaderKey !== 'zone')
+        .map((item) => {
+          const isChecked = this.visibleTableHeaderKeys.includes(item.tableHeaderKey)
+
+          return {
+            ...item,
+            isChecked,
+          }
+        })
+    },
+  },
+
+  watch: {
+    '$route.params.mesh': function () {
+      // Ensures basic state is reset when switching meshes using the mesh selector.
+      this.isLoading = true
+      this.isEmpty = false
+      this.hasError = false
+      this.entityIsLoading = true
+      this.entityIsEmpty = false
+      this.tableDataIsEmpty = false
+
+      this.loadData(0)
+    },
+  },
+
+  created() {
+    const visibleTableHeaderKeys = Storage.get('dpVisibleTableHeaderKeys')
+    if (Array.isArray(visibleTableHeaderKeys)) {
+      this.visibleTableHeaderKeys = visibleTableHeaderKeys
+    }
+  },
+
+  beforeMount() {
+    this.loadData(this.offset)
+  },
+
+  methods: {
+    /**
+     * Ensures that the dropdown menu isn’t toggled whenever a checkbox is checked/unchecked.
+     *
+     * @param {Event} event
+     */
+    stopPropagatingClickEvent(event) {
+      event.stopPropagation()
+    },
+
+    /**
+     * @param {Event} event
+     * @param {string} tableHeaderKey
+     */
+    updateVisibleTableHeaders(event, tableHeaderKey) {
+      const input = /** @type {HTMLInputElement} */ (event.target)
+      const index = this.visibleTableHeaderKeys.findIndex((key) => key === tableHeaderKey)
+
+      if (input.checked && index === -1) {
+        this.visibleTableHeaderKeys.push(tableHeaderKey)
+      } else if (!input.checked && index > -1) {
+        this.visibleTableHeaderKeys.splice(index, 1)
+      }
+
+      Storage.set('dpVisibleTableHeaderKeys', Array.from(new Set(this.visibleTableHeaderKeys)))
+    },
+
+    onCreateClick() {
+      datadogLogs.logger.info(datadogLogEvents.CREATE_DATA_PLANE_PROXY_CLICKED)
+    },
+
+    buildEntity(basicData, tags, dataplaneInsight, versions) {
+      const mtls = dataplaneInsight.mTLS ? parseMTLSData(dataplaneInsight.mTLS) : null
+
+      return { basicData, tags, mtls, versions }
+    },
+
+    getEmptyState() {
+      return {
+        title: 'No Data',
+        message: this.$options.emptyStateMsg,
+      }
+    },
+
+    async parseData(response) {
+      const { dataplane = {}, dataplaneInsight = {} } = response
+      const { name = '', mesh = '' } = response
+      const { subscriptions = [] } = dataplaneInsight
+
+      const nameRoute = {
+        name: 'data-plane-detail-view',
+        params: {
+          mesh,
+          dataPlane: name,
+        },
+      }
+      const meshRoute = {
+        name: 'mesh-child',
+        params: {
+          mesh,
+        },
+      }
+
+      /**
+       * Handle our tag collections based on the dataplane type.
+       */
+      const tags = dpTags(dataplane).filter((tag) => IMPORTANT_DATA_PLANE_TAG_LABELS.includes(tag.label))
```

Comment:
What's the of extracting this as a constant when the 3 entries in your constant are used just after? Seems like this would be better inlined.

Reply:
I’m not sure I’m following.

Do you mean to suggest to look up service, protocol, and zone on `dataplane` directly rather than constructing the tag list and then finding them again?

Reply:
I'm just saying the global var seems not needed here. It's just asking the reader of the code to context switch when it's only used here and the 3 following lines would need to evolve accodingly to this global var.

Reply:
Ah, now I understand. Done.